SimDataFormats/TrackerDigiSimLink/interface/PixelSimHitAddExtraInfo.h

```diff
@@ -0,0 +1,41 @@
+#ifndef PixelSimHitAddExtraInfo_h
+#define PixelSimHitAddExtraInfo_h
+
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include <vector>
+#include <cstdint>
+
+class PixelSimHitAddExtraInfo {
+public:
+  PixelSimHitAddExtraInfo(size_t Hindex, Local3DPoint entryP, Local3DPoint exitP, unsigned int ch) {
```

Comment:
Here it would be more efficient to have `const Local3DPoint& entryP, const Local3DPoint& exitP` to avoid making unecessary copies of `Local3DPoint` objects.

Reply:
Is `unsigned int ch` really needed here given the fact that `void addDigiInfo(unsigned int theDigiChannel)` exists?

Reply:
Concerning the comment on the`unsigned int ch`, I would prefer to keep the 
[logic of filling the information in SiPixelDigitizer.cc](https://github.com/carolinecollard/cmssw/blob/additionalDigiInfo_12_4_X/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.cc#L319:L332). In that case, the addDigiInfo function is only used to add a channel number to the existing list.

Reply:
In the context of your example, my suggestion was to replace
```C++
PixelSimHitExtraInfo newSHEntry(loopNewClass->hitIndex(),
                                loopNewClass->entryPoint(),
                                loopNewClass->exitPoint(),
                                loopNewClass->channel());
```
with
```C++
PixelSimHitExtraInfo newSHEntry(loopNewClass->hitIndex(),
                                loopNewClass->entryPoint(),
                                loopNewClass->exitPoint());
newEntry.addDigiInfo(loopNewClass->channel());
```
so that the channel number only appears in calls to `addDigiInfo` but I guess the logic is such that if an instance of `PixelSimHitExtraInfo` needs to be created, at least one channel number is also needed so it is ok to do a `push_back` already in the constructor.